newa/__init__.py

```diff
@@ -742,7 +747,219 @@ class IssueAction:  # type: ignore[no-untyped-def]
 
 
 @define
-class ErratumConfig(Serializable):
+class ErratumConfig(Serializable):  # type: ignore[no-untyped-def]
+
+    project: str = field()
+    transitions: dict[str, list[str]] = field()
     issues: list[IssueAction] = field(  # type: ignore[var-annotated]
         factory=list, converter=lambda issues: [
             IssueAction(**issue) for issue in issues])
+
+
+@frozen
+class IssueHandler:
+    """ An interface to Jira instance handling a specific ErratumJob """
+
+    erratum_job: ErratumJob = field()
+    url: str = field()
+    token: str = field()
+    project: str = field()
+
+    # Each project can have different semantics of issue status.
+    transitions: dict[str, list[str]] = field()
+
+    # We assume that all projects have the following two custom fields mapped
+    # as follows.
+    custom_field_map: ClassVar[dict[str, str]] = {
+        "field_epic_link": "customfield_12311140",
+        "field_epic_name": "customfield_12311141",
+        }
+
+    # Actual Jira connection.
+    connection: jira.JIRA = field(init=False)
+
+    # Cache of Jira user names mapped to e-mail addresses.
+    user_names: dict[str, str] = field(init=False, default={})
+
+    # NEWA label
+    newa_label: ClassVar[str] = "NEWA"
+
+    @connection.default  # pyright: ignore [reportAttributeAccessIssue]
+    def connection_factory(self) -> jira.JIRA:
+        return jira.JIRA(self.url, token_auth=self.token)
+
+    def newa_id(self, action: IssueAction, partial: bool = False) -> str:
+        """
+        NEWA identifier
+
+        Construct so-called NEWA identifier - it identifies all issues of given
+        action for errata. By default it defines issues related to the current
+        respin. If 'partial' is defined it defines issues relevant for all respins.
+        """
+
+        newa_id = f"{IssueHandler.newa_label} {action.id}: {self.erratum_job.id}"
```

Comment:
I would have to escape # (because Jira interprets it as ordered list). Can we do this?

`::: NEWA errata_epic: E: 130704 @ RHEL-8.8.0.Z.EUS (libreswan-4.9-3.el8_8.1) :::`

Reply:
Works for me.

Reply:
Fixed.